src/server.cpp

```diff
@@ -58,12 +58,31 @@ void Server::setHostname(std::string_view hostname)
 
 const std::string& Server::getFQDN() const noexcept { return m_fqdn; }
 
-/* TODO: Validate if FQDN is in right format */
 void Server::setFQDN(const std::string& fqdn)
 {
     if (fqdn.size() > 255)
         throw std::runtime_error("FQDN cannot be bigger than 255 characters");
 
+    const std::regex fqdnPattern(
```

Comment:
We are missing a documentation of what the regex rule do.

Reply:
It validates whether an FQDN is valid or not.

Reply:
> It validates whether an FQDN is valid or not.

Yes but which cases did you considered?

Reply:
Does it allow starting with a number? We should look at the edge cases, that was I was referring to.

Reply:
@lbgracioso can you add some unit tests for those checkers?